### PR TITLE
Update docs with new changes; Add info about new kanister func

### PR DESCRIPTION
**Pull request by SupriyaKasten**
## Change Overview

Update docs with; add info about new kanister func 

## Pull request type

Please check the type of change your PR introduces:
- [ ] Work in Progress
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Trival/Minor
- [ ] Bugfix
- [ ] Feature
- [x] Documentation

### CODE REVIEW

**Comment by tdmanv** [comment]
```./functions.rst:612: D001 Line too long```

**Comment by SupriyaKasten** [comment]
@tdmanv  my bad!

File: docs/architecture.rst [hunk 1]
```diff
@@ -149,6 +152,7 @@ as follows:
       ConfigMaps map[string]ObjectReference `json:"configMaps"`
       Secrets map[string]ObjectReference    `json:"secrets"`
       Profile *ObjectReference              `json:"profile"`
+      Options map[string]string             `json:"options"`
```
Comment (pavannd1) [inline docs/architecture.rst]: [minor] Please check formatting here. Looks like _italics_ is enabled.

File: docs/functions.rst [hunk 2]
```diff
@@ -290,6 +294,7 @@ Outputs:
    :widths: 5,5,15
 
    `backupTag`,`string`, unique tag added to the backup
+   `backupID`,`string`, unique snapshot id generated by restic during backup
```
Comment (pavannd1) [inline docs/functions.rst]: Let's not use `restic` here. `generated during backup`

File: docs/functions.rst [hunk 3]
```diff
@@ -313,6 +318,63 @@ Example:
             includePath: /mnt/data
             backupArtifactPrefix: s3-bucket/path/artifactPrefix
 
+.. _backupdataall:
+
+BackupDataAll
+-------------
+
+This function concurrently backs up data from more than one pods into an any
```
Comment (pavannd1) [inline docs/functions.rst]: `data from one or more pods`

File: docs/functions.rst [hunk 4]
```diff
@@ -339,15 +401,17 @@ and restores data to the specified path.
    `namespace`, Yes, `string`, namespace in which to execute
    `image`, Yes, `string`, image to be used for running restore
    `backupArtifactPrefix`, Yes, `string`, path to the backup on the object store
-   `backupTag`, Yes, `string`, unique tag added during the backup
+   `backupIdentifier`, No, `string`, (required if backupTag not provided) unique snapshot id generated by restic during backup
```
Comment (pavannd1) [inline docs/functions.rst]: Remove `restic`

File: docs/functions.rst [hunk 5]
```diff
@@ -384,6 +449,81 @@ For this phase, we will use the `backupInfo` Artifact provided by backup functio
...
       replicas: 1
 
+
+.. _restoredataall:
+
+RestoreDataAll
+--------------
+
+This function concurrently restores data backed up by the :ref:`backupdataall`
+function, on more than one pods.
```
Comment (pavannd1) [inline docs/functions.rst]: `one or more pods`

File: docs/functions.rst [hunk 6]
```diff
@@ -384,6 +449,81 @@ For this phase, we will use the `backupInfo` Artifact provided by backup functio
...
+It concurrently runs a job Pod for each workload Pod, that mounts the
+respective PVCs of that Pod and restores data to the specified path.
```
Comment (pavannd1) [inline docs/functions.rst]: [minor] `of that Pod` can be removed

File: docs/functions.rst [hunk 7]
```diff
@@ -416,6 +557,8 @@ Outputs:
    :align: left
    :widths: 5,5,15
 
+   `backupID`,`string`, unique snapshot id generated by restic where data was copied
```
Comment (pavannd1) [inline docs/functions.rst]: Remove `restic`

File: docs/spelling_wordlist.txt [hunk 8]
```diff
@@ -34,3 +34,7 @@ actionset
 objectstore
 Elasticsearch
 backupInfo
+restic
```
Comment (pavannd1) [inline docs/spelling_wordlist.txt]: Can be removed here as well

File: docs/architecture.rst (hunk 1 above)
Comment (pavannd1) [reply docs/architecture.rst]: @SupriyaKasten I think this is because of `*ObjectReference`. * enabled italics.

File: docs/functions.rst [hunk 9]
```diff
@@ -463,11 +608,46 @@ For this phase, we will use the `backupInfo` Artifact provided by backup functio
...
+.. csv-table::
+   :header: "Argument", "Required", "Type", "Description"
+   :align: left
+   :widths: 5,5,5,15
+
+   `namespace`, Yes, `string`, namespace in which to execute
+   `backupArtifactPrefix`, Yes, `string`, path to the backup on the object store
+   `backupInfo`, Yes, `string`, snapshot info generated as output in BackupDataAll function
+   `encryptionKey`, No, `string`, encryption key to be used during backups
+   `reclaimSpace`, NO, `bool`, provides a way to specify if space should be reclaimed
```
Comment (pavannd1) [inline docs/functions.rst]:
```suggestion
   `reclaimSpace`, No, `bool`, provides a way to specify if space should be reclaimed
```

File: docs/architecture.rst (hunk 1 above)
Comment (SupriyaKasten) [reply docs/architecture.rst]: Sure!

**Pull request merged**